cfme/infrastructure/host.py

```diff
@@ -173,70 +81,129 @@ def __init__(self, **kwargs):
             super(Host.Credential, self).__init__(**kwargs)
             self.ipmi = kwargs.get('ipmi')
 
-    def _submit(self, cancel, submit_button):
-        if cancel:
-            sel.click(form_buttons.cancel)
-            # sel.wait_for_element(page.configuration_btn)
-        else:
-            sel.click(submit_button)
-            flash.assert_no_errors()
-
     def create(self, cancel=False, validate_credentials=False):
-        """
-        Creates a host in the UI
+        """Creates a host in the UI.
 
         Args:
-           cancel (boolean): Whether to cancel out of the creation.  The cancel is done
-               after all the information present in the Host has been filled in the UI.
-           validate_credentials (boolean): Whether to validate credentials - if True and the
+           cancel (bool): Whether to cancel out of the creation. The cancel is done after all the
+               information present in the Host has been filled in the UI.
+           validate_credentials (bool): Whether to validate credentials - if True and the
                credentials are invalid, an error will be raised.
         """
-        navigate_to(self, 'Add')
-        fill(properties_form, self._form_mapping(True, **self.__dict__))
-        fill(credential_form, self.credentials, validate=validate_credentials)
-        fill(credential_form, self.ipmi_credentials, validate=validate_credentials)
-        self._submit(cancel, host_add_btn)
+        view = navigate_to(self, "Add")
+        view.fill({
+            "name": self.name,
+            "hostname": self.hostname or self.ip_address,
+            "host_platform": self.host_platform,
+            "custom_ident": self.custom_ident,
+            "ipmi_address": self.ipmi_address,
+            "mac_address": self.mac_address
+        })
+        if self.credentials is not None:
+            view.endpoints.default.fill({
```

Comment:
every credential has view_value_mapping for fill. It would be better to use it here.